### PR TITLE
chore(csv): remove internal dead code

### PR DESCRIPTION
**Pull request by ayame113**
I removed the `readRecord()` in `csv/_io.ts` as it is not used anywhere.

### CODE REVIEW

File: csv/csv_parse_stream.ts [hunk 1]
```diff
@@ -41,8 +41,8 @@ class StreamLineReader implements LineReader {
     }
   }
 
-  isEOF(): Promise<boolean> {
-    return Promise.resolve(this.#done);
+  isEOF(): boolean {
+    return this.#done;
   }
```
Comment (ayame113) [inline csv/csv_parse_stream.ts:46]: Since `isEOF` no longer needs to be async, I made it sync.

File: csv/csv_parse_stream.ts [hunk 2]
```diff
@@ -87,7 +87,7 @@ export class CsvParseStream<
 
   #headers: readonly string[] = [];
 
-  constructor(options: T = defaultReadOptions as T) {
+  constructor(options?: T) {
     this.#options = {
       ...defaultReadOptions,
       ...options,
```
Comment (ayame113) [inline csv/csv_parse_stream.ts:93]: defaultReadOptions is set on line 92, so there is no need to use it as a default value for the constructor here.

**Review by iuioiua — APPROVED** [review]

File: csv/csv_parse_stream.ts [hunk 3]
```diff
@@ -41,8 +41,8 @@ class StreamLineReader implements LineReader {
     }
   }
 
-  isEOF(): Promise<boolean> {
-    return Promise.resolve(this.#done);
+  isEOF(): boolean {
+    return this.#done;
```
Comment (kt3k) [inline csv/csv_parse_stream.ts:45]: Note: `StreamLineReader` (`LineReader` interface) is only used internally in `CsvParseStream`. So we can change this without breaking user facing APIs.

**Review by kt3k — APPROVED** [review]
LGTM

**Pull request merged**